src/parser_TEST.cc

```diff
@@ -59,6 +62,30 @@ sdf::SDFPtr InitSDF()
   return sdf;
 }
 
+/////////////////////////////////////////////////
+/// Checks emitted warnings for custom/unknown elements in log file
+TEST(Parser, CustomUnknownElements)
+{
+  std::string pathBase = PROJECT_SOURCE_PATH;
+  pathBase += "/test/sdf";
+  const std::string path = pathBase +"/custom_and_unknown_elements.sdf";
+
+  sdf::SDFPtr sdf = InitSDF();
+  EXPECT_TRUE(sdf::readFile(path, sdf));
+
+  std::string line;
+  std::string pathHome = getenv("HOME");
```

Comment:
Thanks for adding this test. Is there a way to make it write to a temporary file in the build directory instead of the user's home for testing purposes? Maybe by temporarily setting the `HOME` environment variable? Because otherwise, if the users `sdformat.log` already contains the expected "XML Element[test_unknown]" entry from previous runs, it becomes unclear whether the test is functioning properly.

Reply:
Does it not get overwritten after a new execution? When I was testing it that seemed to be the case. And I put the test function towards the top so that other tests wouldn't output new items in the log file

Reply:
hmm, it does seem to be getting overwritten. That seems like a bug and we probably should not rely on it. I would still recommend writing to a test file so we don't write to anything outside of the build directory. The idea of hermetically sealing our libraries so that they don't write to files outside of the build directory is something we should strive to do as it makes our tests less flaky. It also makes them more portable to other build systems (eg. Bazel) . If manipulating environment variables without affecting other tests is a concern, we could put this in its own `.cc` file and state that no additional tests should be added to the file.

Reply:
Sounds good. So in the test, before `RUN_ALL_TESTS()` is called, https://github.com/osrf/sdformat/blob/95500e0e75f75a1e5ab69e4b264138cbe78f477a/src/parser_TEST.cc#L670 should I set the `HOME` variable to the build directory, then after `RUN_ALL_TESTS()` set it back to what it was originally? It shouldn't affect the other tests in `parser_TEST`

Reply:
Yeah, and I don't think you need to reset it either since the test is done at that point. Setting the environment variable here shouldn't affect the environment variable of the shell.

Reply:
Done 95500e0e75f75a1e5ab69e4b264138cbe78f477a